test/introspection-filter.js

```diff
@@ -0,0 +1,617 @@
+'use strict'
```

Comment:
could we add a test for the gateway? And also upon a gateway refresh?

Reply:
Added a test for the GW too: it works as expected